docs: Convert relative imports to @ aliases in examples

dharun36 opened the pull request:
## For Contributors

### Improving Documentation

This PR improves the Next.js examples by converting relative imports to @ aliases, making them more readable and maintainable.

## Description

Fixes #83717 by converting deep relative imports to clean @ aliases in Next.js examples.

Currently, Next.js examples use long relative imports like:
```typescript
import { i18n, type Locale } from "../../../i18n-config";
```
This PR updates them to use clean @ aliases:
```typescript
import { i18n, type Locale } from "@/i18n-config";
```

## Changes Made

- **Added path alias configuration** `"@/*": ["./*"]` to tsconfig.json files
- **Converted deep relative imports** (d:) to clean @ aliases (`@/`)
- **Updated examples**:
  - `i18n-routing` (specifically mentioned in issue #83717) - 5 files
  - `with-typescript` - 2 files  
  - `with-grafbase` - 2 files
  - `with-mongodb-mongoose` - 2 files

## Benefits

-  **Improved readability**: `@/config` vs `../../../config`
-  **Reduced errors**: No need to count `../` levels when refactoring
-  **Better maintainability**: Easier to move files around
-  **Modern standards**: Aligns with production Next.js practices
-  **Better IDE support**: Enhanced autocomplete and navigation

## Examples of Changes

### Before:
```typescript
import { i18n, type Locale } from "../../../i18n-config";
import { getDictionary } from "../../get-dictionary";
import { grafbase } from "../../../lib/grafbase";
```

### After:
```typescript
import { i18n, type Locale } from "@/i18n-config";
import { getDictionary } from "@/get-dictionary";
import { grafbase } from "@/lib/grafbase";
```

## Testing

-  TypeScript compilation works correctly
-  @ imports resolve properly in IDE
-  Runtime functionality preserved
-  No breaking changes to example functionality

This change is purely a developer experience improvement and doesn't affect runtime behavior.

rdharun36 committed:
docs: Convert relative imports to @ aliases in examples

- Add path alias configuration '@/*': ['./*'] to tsconfig.json files
- Convert deep relative imports (../../../) to clean @ aliases
- Update i18n-routing example (specifically mentioned in issue #83717)
- Update with-typescript, with-grafbase, and with-mongodb-mongoose examples
- Improve code readability and maintainability
- Align with modern Next.js best practices

Fixes #83717

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: 1914b15e9df2acb46cbb9a4e467f8ea39ff1fece

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

icyJoseph reviewed (commented):
- [x] examples/i18n-routing: fails because of layout `lang: Locale` typing - typed helpers issue / also out of data with tsconfig's change to `jsx` 
    - but OK to merge I guess, broken for unrelated reasons
- [x] examples/with-grafbase: fails because /posts/[slug]/page.tsx doesn't use async param access
    - same, OK to merge I guess, broken for unrelated reasons
- [x] examples/with-mongodb-mongoose: build fails, and not all paths have been converted
    - the build fails because of unrelated reasons though (I think)
```console
examples/with-mongodb-mongoose/pages/[id]/edit.tsx
3:import Form from "../../components/Form";

examples/with-mongodb-mongoose/pages/[id]/index.tsx
4:import dbConnect from "../../lib/dbConnect";
5:import Pet, { Pets } from "../../models/Pet";
```

- [x] examples/with-typescript: builds OK

rdharun36 committed:
docs: Add more @ import conversions in with-mongodb-mongoose

- Convert Form import from ../../components/Form to @/components/Form in edit.tsx
- Convert dbConnect and Pet imports from ../../ to @/ in index.tsx
- Further improve code readability and maintainability
- Complete coverage of relative imports in mongodb-mongoose example

Related to #83717

rdharun36 committed:
fix: Correct path mapping in with-mongodb-mongoose tsconfig.json

- Change '@/*': ['./src/*', './lib/*'] to '@/*': ['./*']
- Enable @ imports to resolve to all root-level directories
- Fix @/components/Form import resolution issue
- Ensure consistency with other examples' path configuration

Related to #83717

dharun36 committed:
Merge branch 'canary' into fix/examples-use-at-imports

dharun36 committed:
Merge branch 'canary' into fix/examples-use-at-imports

icyJoseph commented:
Ty for the PR, I'll give it one more review and we'll see if we need to run the checks, I don't think they are necessary for examples. Likely gonna merge during the day.

icyJoseph commented:
@dharun36 one more thing, please run:

```console
❯ pnpm prettier-check 

> nextjs-project@0.0.0 prettier-check 
> prettier --check .

Checking formatting...
[warn] examples/i18n-routing/tsconfig.json
[warn] examples/with-grafbase/tsconfig.json
[warn] examples/with-mongodb-mongoose/tsconfig.json
[warn] examples/with-typescript/tsconfig.json
[warn] Code style issues found in 5 files. Run Prettier with --write to fix.
```

Verify that you get those issues, and then run `pnpm prettier-fix` - 

The command will print every file it visits, but it should only change those files. 

Commit and push the fixed formats.

rdharun36 committed:
style: Fix prettier formatting for tsconfig.json files

- Run prettier --write on all modified tsconfig.json files
- Fix code style issues found in 4 files
- Ensure consistent formatting as requested by maintainer

Related to #83717

rdharun36 committed:
Merge remote-tracking branch 'fork/fix/examples-use-at-imports'

dharun36 committed:
Merge branch 'canary' into fix/examples-use-at-imports

icyJoseph committed:
Merge branch 'canary' into fix/examples-use-at-imports
